src/relayer/Relayer.sol

```diff
@@ -70,36 +69,40 @@ contract Relayer is Guarded, IRelayer {
         // We always update the oracles before retrieving the rates
         bool oracleUpdated = IOracle(oracle).update();
         (int256 oracleValue, bool isValid) = IOracle(oracle).value();
-        if (oracleUpdated && isValid) {
-            // If the change in delta rate from the last update is bigger than the threshold value push
-            // the rates to Collybus
-            if (
-                checkDeviation(
-                    _lastUpdateValue,
-                    oracleValue,
-                    minimumPercentageDeltaValue
-                )
-            ) {
-                _lastUpdateValue = oracleValue;
-
-                if (relayerType == RelayerType.DiscountRate) {
-                    ICollybus(collybus).updateDiscountRate(
-                        uint256(encodedTokenId),
-                        uint256(oracleValue)
-                    );
-                } else if (relayerType == RelayerType.SpotPrice) {
-                    ICollybus(collybus).updateSpot(
-                        address(uint160(uint256(encodedTokenId))),
-                        uint256(oracleValue)
-                    );
-                }
-
-                emit UpdatedCollybus(
-                    encodedTokenId,
-                    uint256(oracleValue),
-                    relayerType
+
+        // If the oracle was not updated or the value was invalid then we can exit early as we will not push data to Collybus
+        if(!oracleUpdated || !isValid){
+            return oracleUpdated;
+        }
+        
+        // If the change in delta rate from the last update is bigger than the threshold value push
+        // the rates to Collybus
+        if (
+            checkDeviation(
+                _lastUpdateValue,
+                oracleValue,
+                minimumPercentageDeltaValue
+            )
+        ) {
```

Comment:
Similarly to the negation above, an early exit can be done here.

de Morgan's laws can be used
https://en.wikipedia.org/wiki/De_Morgan%27s_laws